src/drivers/px4fmu/fmu.cpp

```diff
@@ -1413,7 +1413,11 @@ PX4FMU::cycle()
 #endif
 		/* check arming state */
 		bool updated = false;
-		orb_check(_armed_sub, &updated);
+
+		// only update params if disarmed
+		if (!_armed.armed) {
+			orb_check(_armed_sub, &updated);
```

Comment:
This looks potentially wrong - why are we only updating the armed state if we are disarmed? Generally we want PWM_REV and PWM_TRIM params to update also when armed, so if unsure this should be just dropped entirely.

Reply:
The idea was to skip changing PWM parameters when flying, which could easily be unsafe (and a small performance optimization).